#
# Submitted-By: Sascha Hauer, 2006-03-22
# Committed-By: Sascha Hauer
#
# Error:
#
# Xserver does not start when CAN devices are present
#
# Description:
#
# the xserver uses getifaddrs(3) to get a list of network
# interfaces present in the system. It then dereferences
# struct sockaddr  *ifa_addr
# without checking a NULL pointer before.
# The manpage of getifaddrs says that this pointer can be
# NULL, so this is a bug in Xorg
#
# State:
# 
# pending upstream
#
Index: xorg-server-X11R7.0-1.0.1/os/access.c
===================================================================
--- xorg-server-X11R7.0-1.0.1.orig/os/access.c
+++ xorg-server-X11R7.0-1.0.1/os/access.c
@@ -1858,6 +1858,8 @@ ConvertAddr (
 {
     if (*len == 0)
         return (FamilyLocal);
+    if (!saddr)
+        return -1;
     switch (saddr->sa_family)
     {
     case AF_UNSPEC:
